### ISSUE TITLE
tparse can return status code 0 although "go test" returned non-zero

### ISSUE THREAD

**Issue by marco-m** [mfridman/tparse#54]
Hello,
thanks for tparse!

the tparse README shows as the first example of usage:

    go test fmt -json | tparse -all

I found that when a Go module has tests for at least two different packages, and one of the tests has a compilation error, `go test` returns a non-zero status code as expected, while `go test . -json | tparse` will return a zero status code.

What I suggest is to consider amending the example as follows:

    set -o pipefail && go test fmt -json | tparse -all

I made a as-small-as-possible repro at http://github.com/marco-m/tparse-bugs.

**Comment by mfridman** [comment]
Thanks for providing a repo, this looks totally reasonable .. I'll update the README.

But I'd like to also take a look and see why that's the case. The exit code tparse computes must match that of `go test` .. afaik there isn't a way to stream the output into tparse and then read the status code of the `go test` command at the end.

**Comment by mfridman** [comment]
Ye, this is definitely a bug and an excellent issue. I've updated the README as suggested, but have also started working on a fix.

tl;dr `tparse` needs to detect the special case `^FAIL`, which iirc happens for build and setup failures. Unfortunately, this output doesn't make it into the json output and instead needs to be special-cased as the lines are being parsed.

**Comment by mfridman** [comment]
Thank you for such a good reproducible example. I found a solution I'm fairly happy with and will push a fix shortly.

Running the provided example should now output the correct status code (2) and also surface the package that failed. `tparse` now detects both `[build failed]` and `[setup failed]`

```
FAIL    github.com/marco-m/tparse-bugs [build failed]
=== RUN   TestB
--- PASS: TestB (0.00s)
PASS
ok      github.com/marco-m/tparse-bugs/b        0.320s
┌────────────────────────────────────────────────────────────────────────────────────┐
│  STATUS │ ELAPSED │             PACKAGE              │ COVER │ PASS │ FAIL │ SKIP  │
│─────────┼─────────┼──────────────────────────────────┼───────┼──────┼──────┼───────│
│  FAIL   │ 0.00s   │ github.com/marco-m/tparse-bugs   │ --    │ --   │ --   │ --    │
│         │         │ [build failed]                   │       │      │      │       │
│  PASS   │ 0.32s   │ github.com/marco-m/tparse-bugs/b │ 0.0%  │    1 │    0 │    0  │
└────────────────────────────────────────────────────────────────────────────────────┘
status code: 2
```

**Comment by marco-m** [comment]
Thanks, very nice!

**Comment by marco-m** [comment]
Is `pipefail` still needed? If not, it could be removed from the README.

**Comment by marco-m** [comment]
@mfridman ping in case you missed my previous comment, since this issue is closed:

> Is pipefail still needed? If not, it could be removed from the README.
